### PR TITLE
ci: fix can't restore cache

### PR DESCRIPTION
**Pull request by qwq233**

### CODE REVIEW

**Review by NextAlone — APPROVED** [review]

**Review by NextAlone — DISMISSED** [review]

**Review by NextAlone — APPROVED** [review]
不是升级AGP有一个致命问题吗（来源ggh）

File: app/build.gradle [hunk 1]
```diff
@@ -12,7 +12,7 @@ if (performSigning) {
 }
 
 android {
-    compileSdkVersion 30
+    compileSdk 31
     buildToolsVersion "30.0.3"
```
Comment (NextAlone) [inline app/build.gradle:16]: need not to update? and targetSdkVersion
```suggestion
    buildToolsVersion "31.0.0"
```

**Review by NextAlone — CHANGES_REQUESTED** [review]

**Review by NextAlone — CHANGES_REQUESTED** [review]
```
E/QNdump: java.lang.UnsatisfiedLinkError: Unsupported ABI: armeabi-v7a
        at nil.nadph.qnotified.util.Natives.extractNativeLibrary(Natives.java:189)
        at nil.nadph.qnotified.util.Natives.load(Natives.java:148)
        at nil.nadph.qnotified.startup.StartupRoutine.execPostStartupInit(StartupRoutine.java:57)
        at nil.nadph.qnotified.startup.StartupHook.execStartupInit(StartupHook.java:76)
        at nil.nadph.qnotified.startup.StartupHook$1.afterHookedMethod(StartupHook.java:197)
        at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:136)
        at LspHooker_.doStep(Unknown Source:8)
        at com.tencent.mobileqq.startup.step.Step.step(P:126)
        at com.tencent.mobileqq.startup.step.Step.doStep(P:158)
        at com.tencent.mobileqq.startup.step.Step.step(P:126)
        at com.tencent.mobileqq.startup.director.StartupDirector.a(P:428)
        at com.tencent.common.app.BaseApplicationImpl.onCreate(P:336)
        at com.tencent.mobileqq.qfix.QFixApplication.onCreate(P:30)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1226)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7016)
        at java.lang.reflect.Method.invoke(Native Method)
        at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:107)
        at LspHooker_.handleBindApplication(Unknown Source:11)
        at android.app.ActivityThread.access$1700(ActivityThread.java:253)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2180)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:233)
        at android.os.Looper.loop(Looper.java:344)
        at android.app.ActivityThread.main(ActivityThread.java:8171)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
```

**Comment by qwq233** [comment]
> ```
> E/QNdump: java.lang.UnsatisfiedLinkError: Unsupported ABI: armeabi-v7a
>         at nil.nadph.qnotified.util.Natives.extractNativeLibrary(Natives.java:189)
>         at nil.nadph.qnotified.util.Natives.load(Natives.java:148)
>         at nil.nadph.qnotified.startup.StartupRoutine.execPostStartupInit(StartupRoutine.java:57)
>         at nil.nadph.qnotified.startup.StartupHook.execStartupInit(StartupHook.java:76)
>         at nil.nadph.qnotified.startup.StartupHook$1.afterHookedMethod(StartupHook.java:197)
>         at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:136)
>         at LspHooker_.doStep(Unknown Source:8)
>         at com.tencent.mobileqq.startup.step.Step.step(P:126)
>         at com.tencent.mobileqq.startup.step.Step.doStep(P:158)
>         at com.tencent.mobileqq.startup.step.Step.step(P:126)
>         at com.tencent.mobileqq.startup.director.StartupDirector.a(P:428)
>         at com.tencent.common.app.BaseApplicationImpl.onCreate(P:336)
>         at com.tencent.mobileqq.qfix.QFixApplication.onCreate(P:30)
>         at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1226)
>         at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7016)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:107)
>         at LspHooker_.handleBindApplication(Unknown Source:11)
>         at android.app.ActivityThread.access$1700(ActivityThread.java:253)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2180)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loopOnce(Looper.java:233)
>         at android.os.Looper.loop(Looper.java:344)
>         at android.app.ActivityThread.main(ActivityThread.java:8171)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
> ```

你这怎么触发的 我adb logcat没看到

**Comment by NextAlone** [comment]
@qwq233 直接hook不上，缺库

**Comment by qwq233** [comment]
> @qwq233 直接hook不上，缺库，rebase on your branch and build， 你没问题吗？

我操作问题？

```
❯ adb logcat|grep -E 'QNdump|NADump|NAdump'
11-07 09:15:57.646 20894 20894 D QNdump  : hookProc=0xdb590333
11-07 09:15:57.646 20894 20894 D QNdump  : dlopen by extract success
11-07 09:16:04.765 21288 21288 D QNdump  : hookProc=0xdb590333
11-07 09:16:04.766 21288 21288 D QNdump  : dlopen by extract success
11-07 09:16:10.952 21546 21546 D QNdump  : hookProc=0xdb590333
11-07 09:16:10.953 21546 21546 D QNdump  : dlopen by extract success
11-07 09:16:22.415 21911 21911 D QNdump  : hookProc=0xdb590333
11-07 09:16:22.415 21911 21911 D QNdump  : dlopen by extract success
11-07 09:16:22.522 21911 21911 D QNdump  : injectModuleResources: Resources injection success
11-07 09:16:22.663 21911 21911 D QNdump  : injectModuleResources: Resources injection success
11-07 09:16:24.446 21911 21947 E QNdump  : java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
11-07 09:16:24.446 21911 21947 E QNdump  :      at de.robv.android.xposed.XposedBridge.hookMethod(Unknown Source:3)
11-07 09:16:24.446 21911 21947 E QNdump  :      at cc.ioctl.hook.RoundAvatarHook.initOnce(RoundAvatarHook.java:103)
11-07 09:16:24.446 21911 21947 E QNdump  :      at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:136)
11-07 09:16:24.446 21911 21947 E QNdump  :      at LspHooker_.doStep(Unknown Source:8)
11-07 09:16:24.446 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.step(P:126)
11-07 09:16:24.446 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.doStep(P:158)
11-07 09:16:24.446 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.step(P:126)
11-07 09:16:24.446 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.director.StartupDirector.a(P:788)
11-07 09:16:24.446 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.director.StartupDirector.handleMessage(P:801)
11-07 09:16:24.446 21911 21947 E QNdump  :      at android.os.Handler.dispatchMessage(Unknown Source:12)
11-07 09:16:24.446 21911 21947 E QNdump  :      at android.os.Looper.loop(Unknown Source:249)
11-07 09:16:24.446 21911 21947 E QNdump  :      at android.os.HandlerThread.run(Unknown Source:28)
11-07 09:16:24.591 21911 21947 E QNdump  : java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.reflect.Method[] java.lang.Class.getDeclaredMethods()' on a null object reference
11-07 09:16:24.591 21911 21947 E QNdump  :      at me.kyuubiran.hook.RemoveQbossAD.initOnce(RemoveQbossAD.java:48)
11-07 09:16:24.591 21911 21947 E QNdump  :      at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:136)
11-07 09:16:24.591 21911 21947 E QNdump  :      at LspHooker_.doStep(Unknown Source:8)
11-07 09:16:24.591 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.step(P:126)
11-07 09:16:24.591 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.doStep(P:158)
11-07 09:16:24.591 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.step.Step.step(P:126)
11-07 09:16:24.591 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.director.StartupDirector.a(P:788)
11-07 09:16:24.591 21911 21947 E QNdump  :      at com.tencent.mobileqq.startup.director.StartupDirector.handleMessage(P:801)
11-07 09:16:24.591 21911 21947 E QNdump  :      at android.os.Handler.dispatchMessage(Unknown Source:12)
11-07 09:16:24.591 21911 21947 E QNdump  :      at android.os.Looper.loop(Unknown Source:249)
11-07 09:16:24.591 21911 21947 E QNdump  :      at android.os.HandlerThread.run(Unknown Source:28)
11-07 09:16:24.842 22002 22002 D QNdump  : hookProc=0xdb590333
11-07 09:16:24.842 22002 22002 D QNdump  : dlopen by extract success
11-07 09:16:26.599 21911 21911 D QNdump  : NA: Throwable: java.lang.ClassCastException: com.tencent.mobileqq.activity.home.MainFragment cannot be cast to android.view.ViewGroup
11-07 09:16:26.599 21911 21911 D QNdump  :      at xyz.nextalone.hook.ChatWordsCount$initOnce$1$1.invoke(ChatWordsCount.kt:83)
11-07 09:16:26.599 21911 21911 D QNdump  :      at xyz.nextalone.hook.ChatWordsCount$initOnce$1$1.invoke(ChatWordsCount.kt:82)
11-07 09:16:26.599 21911 21911 D QNdump  :      at xyz.nextalone.util.HookUtilsKt$hookBeforeAllConstructors$1.beforeHookedMethod(HookUtils.kt:355)
11-07 09:16:26.599 21911 21911 D QNdump  :      at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:75)
11-07 09:16:26.599 21911 21911 D QNdump  :      at LspHooker_.constructor(Unknown Source:17)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.t(P:3846)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.T(P:3954)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.d(P:463)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation$UICallBack.b(P:1023)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation$UICallBack.handleMessage(P:863)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.tencent.mobileqq.utils.CustomHandler.handleMessage(P:25)
11-07 09:16:26.599 21911 21911 D QNdump  :      at mqq.os.MqqHandler.dispatchMessage(P:99)
11-07 09:16:26.599 21911 21911 D QNdump  :      at mqq.os.MqqMessageQueue.dequeue(P:294)
11-07 09:16:26.599 21911 21911 D QNdump  :      at mqq.os.MqqMessageQueue.queueIdle(P:251)
11-07 09:16:26.599 21911 21911 D QNdump  :      at android.os.MessageQueue.next(Unknown Source:161)
11-07 09:16:26.599 21911 21911 D QNdump  :      at android.os.Looper.loop(Unknown Source:77)
11-07 09:16:26.599 21911 21911 D QNdump  :      at android.app.ActivityThread.main(Unknown Source:101)
11-07 09:16:26.599 21911 21911 D QNdump  :      at java.lang.reflect.Method.invoke(Native Method)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
11-07 09:16:26.599 21911 21911 D QNdump  :      at com.android.internal.os.ZygoteInit.main(Unknown Source:313)
11-07 09:16:27.015 21911 21911 D QNdump  : NA: Throwable: java.lang.ClassCastException: com.tencent.mobileqq.activity.home.MainFragment cannot be cast to android.view.ViewGroup
11-07 09:16:27.015 21911 21911 D QNdump  :      at xyz.nextalone.hook.ChatWordsCount$initOnce$1$2.invoke(ChatWordsCount.kt:92)
11-07 09:16:27.015 21911 21911 D QNdump  :      at xyz.nextalone.hook.ChatWordsCount$initOnce$1$2.invoke(ChatWordsCount.kt:90)
11-07 09:16:27.015 21911 21911 D QNdump  :      at xyz.nextalone.util.HookUtilsKt$hookAfterAllConstructors$1.afterHookedMethod(HookUtils.kt:364)
11-07 09:16:27.015 21911 21911 D QNdump  :      at de.robv.android.xposed.LspHooker.handleHookedMethod(Unknown Source:136)
11-07 09:16:27.015 21911 21911 D QNdump  :      at LspHooker_.constructor(Unknown Source:17)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.t(P:3846)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.T(P:3954)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation.d(P:463)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation$UICallBack.b(P:1023)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.activity.home.Conversation$UICallBack.handleMessage(P:863)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.tencent.mobileqq.utils.CustomHandler.handleMessage(P:25)
11-07 09:16:27.015 21911 21911 D QNdump  :      at mqq.os.MqqHandler.dispatchMessage(P:99)
11-07 09:16:27.015 21911 21911 D QNdump  :      at mqq.os.MqqMessageQueue.dequeue(P:294)
11-07 09:16:27.015 21911 21911 D QNdump  :      at mqq.os.MqqMessageQueue.queueIdle(P:251)
11-07 09:16:27.015 21911 21911 D QNdump  :      at android.os.MessageQueue.next(Unknown Source:161)
11-07 09:16:27.015 21911 21911 D QNdump  :      at android.os.Looper.loop(Unknown Source:77)
11-07 09:16:27.015 21911 21911 D QNdump  :      at android.app.ActivityThread.main(Unknown Source:101)
11-07 09:16:27.015 21911 21911 D QNdump  :      at java.lang.reflect.Method.invoke(Native Method)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
11-07 09:16:27.015 21911 21911 D QNdump  :      at com.android.internal.os.ZygoteInit.main(Unknown Source:313)
11-07 09:16:31.164 22535 22535 D QNdump  : hookProc=0xdb590333
11-07 09:16:31.164 22535 22535 D QNdump  : dlopen by extract success
11-07 09:16:33.148 22644 22644 D QNdump  : hookProc=0xdb590333
11-07 09:16:33.148 22644 22644 D QNdump  : dlopen by extract success
```

**Review by keta1 — APPROVED** [review]

**Review by NextAlone — APPROVED** [review]

**Pull request merged**